Tarea 2

H-Cote opened the pull request:
Esta es la primera versión de mi tarea 2. 
No estoy seguro de que el código de la derivada compleja que hice sea correcto. En la derivada simétrica, la gráfica del error me genera dudas sobre el comportamiento del error respecto a h. Por tales motivos, no me fue posible contestar las preguntas. 

Saludos

H-Cote committed:
Agrego las capturas de pantalla de la tarea0

H-Cote committed:
Puse las capturas de pantalla en un documento de jupyter

H-Cote committed:
Merge branch 'master' of https://github.com/lbenet/2017-2_TSFisComputacional-1 into tarea0

H-Cote committed:
Primera versión de la tarea 1

Falta poner códigos como funciones y completar los incisos c) y d) del ejercicio 2

H-Cote committed:
Nuevas notas del curso

Merge branch 'master' of https://github.com/lbenet/2017-2_TSFisComputacional-1 into tarea_1

H-Cote committed:
Puse los cógigos como funciones donde era necesario.
En ejercicio 1 modifiqué los resultados de salida para que la función sea
estable según el tipo y documenté la función. Además, cambié el valor de c por el que se pide.
En el ejercicio 2, aún falta completar los incisos c) y d). La función varianza_rec aún no funciona.
En el ejercicio 3, modifiqué el código del método de Simpson para tener una mejor aproximación al valor de la integral.

H-Cote committed:
Tercera versión de la tarea 1

Las funciones devuelven los resultados en lugar de imprimirlos.

Se completan lo incisos c) y d).

Modificaciones a los códigos de la fórmula general y la varianza para hacerlos más rápidos.

H-Cote committed:
actualización del repositorio
Merge branch 'master' of https://github.com/lbenet/2017-2_TSFisComputacional-1 into tarea_1

H-Cote committed:
Cuarta versión de la tarea 1

En el ejercicio 1, se hizo una modificación para el caso de una sola raíz y cuando a = 0. Falta el caso cuando el valor de ac es pequeño.

En los ejercicio 2 y 3, se documentaron las funciones.

En el ejercicio 2, en el inciso d), la función varianza_promedio_rec() devuelve tanto el promedio como la varianza de la siguiente forma ([varianza],[promedio]), cuyos valores son cercanos a los obtenidos con las funciones mean() y var()

H-Cote committed:
Quinta versión de la tarea 1.
En el ejercicio 1, se cambió la fórmula general habitual por la de fórmula de Citardauq para tener estabilidad de tipo cuando el producto a*c es pequeño.
En el ejercicio 2, en la función varianza_promedio_rec, se sacó la linea del cálculo de la varianza del ciclo for para hacer más rápido los cálculos, sin embargo la función sigue siendo lenta en comparación con las otras.

H-Cote committed:
Sexta versión de la tarea 1.
En el ejercicio 1, se utilizaron la fórmula general junto con la fórmula de Citardauq, para calcular las raices. Esto con el fin de evitar errores cuando el producto de a*c es pequeño. Además, se incluyó como resultado de salida un error cuando el valor de a es igual a cero.

H-Cote committed:
Primera versión de la tarea 2

Falta varias de las preguntas pues no comprendo totalmente el comportamiento del error respecto a h.
No estoy seguro de que la función para la derivada compleja esté bien hecha.

lbenet commented:
Recibida. Más allá de si están las cosas bien o mal (la revisaré después) te hago notar el mensaje que aparece: "This branch cannot be rebased due to conflicts". Esto se debe a que estás cambiando tu Tarea1 y la Tarea2; no debes cambiar ya la Tarea1, o si así lo quieres, mandas otro PR.

lbenet commented:
Comentarios:

-No es necesario que hagas el cálculo de la derivada usando límites.

- En la pregunta 1, es útil que definas a mano el vector con los valores de las `h`, pero es mejor si Julia lo hace por ti! En cuanto a los `BigFloat`s, es algo útil pero alenta mucho las cosas. Ganar precisión es tener más dígitos significativos; el chiste es ganar exactitud, o sea, más dígitos correctos. Esto es algo parecido a lo que le escribí en [este comentario](https://github.com/lbenet/2017-2_TSFisComputacional-1/pull/29#issuecomment-287085023) a Carlos. Usa (en `Float64` valores de `h` aún más pequeños; trata de llegar a 10^{-15}, por ejemplo.

- Los comentarios anteriores esencialmente se aplican al ejercicio 2.

- En el ejercicio 3 te falta *sólo* quedarte con la parte imaginaria (fíjate en lo que sale en la celda [1]]); `imag` es una instrucción útil; y considera valores más chicos de `h`.

lbenet commented:
... Y falta que logres hacer que tus cambios no generen conflictos!

H-Cote committed:
Segunda versión de la tarea 2.

Se hizo un nuevo vector_h con a partir de dos vectores cuyos valores están en el intervalo [0,1). Uno de esto vectores contiene valores muy pequeños y el otro valores un tanto más grandes.

Se retiró la función BigFloat() de todas las funciones.

En la función para la derivada compleja se agregó la función imag() para que devuelva la parte real del resultado.

Aún falta contestar las preguntas.

lbenet commented:
Respecto a la generación de un vector con suficientes elementos, entre cero y uno, te recomiendo mejor usar `linspace` o `logspace` dependiendo si quieres que el espaciamiento sea lineal o logarítmico. Esto te ayudará a hacer una mejor gráfica del error, ya que puedes controlar mejor el número de puntos y su espaciamiento. Esto mismo se aplica a la pregunta 2 y a la 3.

lbenet commented:
Sigue habiendo conflictos que **no** permitirán hacer el merge!

H-Cote committed:
Tercera versión de la tarea 2.

Se genera el vector_h por medio de la función logspace con 1000 entradas entre 0 y 1.
Se contestan las preguntas planteadas.
Las gráficas son coloreadas y se les agragan etiquetas (leyendas).

lbenet commented:
- En la pregunta 1, en tus comentarios finales dice "dismunuye"; la segunda frase la puedes escribir mejor.

- Estoy parcialmente de acuerdo con lo que comentas en la pregunta 4. Lo que hace numéricamente robusta a la deriada compleja es, como dices, que no involucra diferencias, mientras que las otras sí requieren de una derivada; no creo que lo que mencionas de las sucesiones de Cauchy es relevante, pero si lo argumentas mejor estoy dispuesto a aceptarlo. 

- Te falta contestar, usando tus gráficas, como (a priori) convergen las distintas funciones respecto al valor exacto; en principio sólo basta con comparar las pendientes en la escala log-log.

lbenet commented:
Otra cosa: **no** puedo hacer el merge de tu Tarea ya que hay conflictos. Sugiero que, cuando tengas el notebook final, simplemente abras otra rama (a partir de master), ahí copies *únicamente* tu Tarea2, y hagas el PR.

lbenet commented:
Esto continúa en #41, por lo que lo cierro.
